pkg/sessions/persistence/ticket.go

```diff
@@ -0,0 +1,226 @@
+	if err != nil {
+		return fmt.Errorf("failed to encode the session state with the ticket: %v", err)
+	}
+	return saver(t.TicketID, ciphertext, t.Options.Expire)
+}
+
+// LoadSession loads a session from the disk store via the passed LoadFunc
+// using the TicketID as the key. It then decodeds the SessionState using
+// Ticket.Secret to make the AES-GCM cipher.
+//
+// TODO (@NickMeves): Remove legacyV5LoadSession support in V7
+func (t *Ticket) LoadSession(loader LoadFunc) (*sessions.SessionState, error) {
+	ciphertext, err := loader(t.TicketID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load the session state with the ticket: %v", err)
+	}
+	c, err := t.makeCipher()
+	if err != nil {
+		return nil, err
+	}
+	ss, err := sessions.DecodeSessionState(ciphertext, c, false)
+	if err != nil {
+		return t.legacyV5LoadSession(ciphertext)
+	}
+	return ss, nil
+}
+
+// ClearSession uses the passed ClearFunc to delete a session stored with a
+// key of TicketID
+func (t *Ticket) ClearSession(clearer ClearFunc) error {
+	return clearer(t.TicketID)
+}
+
+// SetCookie sets the encoded Ticket as a cookie
+func (t *Ticket) SetCookie(rw http.ResponseWriter, req *http.Request, s *sessions.SessionState) {
+	ticketCookie := t.makeCookie(
+		req,
+		t.EncodeTicket(),
+		t.Options.Expire,
+		*s.CreatedAt,
+	)
+
+	http.SetCookie(rw, ticketCookie)
+}
+
+// ClearCookie removes any cookies that would be where this Ticket would set
+// them
+func (t *Ticket) ClearCookie(rw http.ResponseWriter, req *http.Request) {
+	clearCookie := t.makeCookie(
+		req,
+		"",
+		time.Hour*-1,
+		time.Now(),
+	)
+	http.SetCookie(rw, clearCookie)
+}
+
+// makeCookie makes a cookie, signing the value if present
+func (t *Ticket) makeCookie(req *http.Request, value string, expires time.Duration, now time.Time) *http.Cookie {
+	if value != "" {
+		value = encryption.SignedValue(t.Options.Secret, t.Options.Name, []byte(value), now)
+	}
+	return cookies.MakeCookieFromOptions(
+		req,
+		t.Options.Name,
+		value,
+		t.Options,
+		expires,
+		now,
+	)
+}
+
+// makeCipher makes a AES-GCM cipher out of the Ticket.Secret
+func (t *Ticket) makeCipher() (encryption.Cipher, error) {
+	c, err := encryption.NewGCMCipher(t.Secret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to make an AES-GCM cipher from the ticket secret: %v", err)
+	}
+	return c, nil
+}
+
+// legacyV5LoadSession loads a Redis session created in V5 with historical logic
+//
+// TODO (@NickMeves): Remove in V7
+func (t *Ticket) legacyV5LoadSession(resultBytes []byte) (*sessions.SessionState, error) {
+	block, err := aes.NewCipher(t.Secret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a legacy AES-CFB cipher from the ticket secret: %v", err)
+	}
+
+	stream := cipher.NewCFBDecrypter(block, t.Secret)
+	stream.XORKeyStream(resultBytes, resultBytes)
+
+	cfbCipher, err := encryption.NewCFBCipher(encryption.SecretBytes(t.Options.Secret))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Is it not better to initialise this once at manager creation as we did before?

Reply:
Even when I left in the CookieCipher on the store level in Redis in my original commit - I was still manually creating one here for the legacy use case.

Otherwise I would've needed to propagate a made cipher all the way from any new store implementations to tickets. I was thinking I didn't want to force any new stores to make a dummy cipher just for legacy Redis V5's support.

Plus it allowed me to keeping the Manager/Ticket to just needing the Cookie options passed down from any store, keeping it simple.